application/src/objects/obstacles/obstacle-factory.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var _          = require('underscore');
+var ClownNose  = require('./clown-nose');
+var DynamicBox = require('./dynamic-box');
+var StaticBox  = require('./static-box');
+
+var ObstacleFactory = function(state) {
+    this.state      = state;
+};
+
+ObstacleFactory.prototype.types = {
+    'ClownNose'  : ClownNose,
+    'DynamicBox' : DynamicBox,
+    'StaticBox'  : StaticBox
+};
+
+ObstacleFactory.prototype.loadAssets = function(types)
+{
+    _.each(types, function(type) {
+        if (this.types[type]) {
+            this.state.load.image(type, this.types[type].prototype.getSpritePath());
+        } else {
+            throw new Error('Attempted to load assets for unknown class ' + type);
+        }        
+    }, this);
+};
+
+ObstacleFactory.prototype.getNew = function(type, x, y, collisionGroup)
```

Comment:
Yeah, I only went with Sprite because that's what kind of object is being returned. It might just be that Phaser has a too-broad definition of a sprite. I can go with `getNew`.